src/pgexporter_ext/lib.c

```diff
@@ -485,17 +485,20 @@ os_info(Tuplestorestate* tupstore, TupleDesc tupdesc)
       nulls[OS_INFO_DOMAIN_NAME] = true;
    }
 
-   os_info_file = fopen("/etc/system-release", "r");
+   os_info_file = fopen("/etc/os-release", "r");
 
    if (!os_info_file)
    {
       nulls[OS_INFO_NAME] = true;
    }
    else
    {
-      if (fgets(&buffer[0], max_length, os_info_file) != NULL)
+      while (fgets(&buffer[0], max_length, os_info_file) != NULL)
```

Comment:
As per [this article](https://www.freedesktop.org/software/systemd/man/latest/os-release.html#PRETTY_NAME=), a fall back value must be provided which is "Linux" as it is an optional field, or like similar to your case, it will be empty and will cause problems.

Also, just fyi @jesperpedersen, this might not contain the version for some distros (like in my EndeavourOS, which is a rolling release) as pointed by [the same article](https://www.freedesktop.org/software/systemd/man/latest/os-release.html#PRETTY_NAME=).